sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2203,13 +2220,41 @@ class Analyzer(
       }.nonEmpty)
     }
 
-    private def trimAlias(expr: NamedExpression): Expression = expr match {
+    private def trimAlias(expr: Expression): Expression = expr match {
       case UnresolvedAlias(child, _) => child
       case Alias(child, _) => child
       case MultiAlias(child, _) => child
       case _ => expr
     }
 
+    private def createGenerate(
+        generator: Generator,
+        outer: Boolean,
+        names: Seq[String],
+        child: LogicalPlan): Generate = {
+      Generate(
+        generator,
+        unrequiredChildIndex = Nil,
+        outer = outer,
+        qualifier = None,
+        generatorOutput = ResolveGenerate.makeGeneratorOutput(generator, names),
+        child)
+    }
+
+    private def collectAdjacentGenerators(children: Seq[Expression]): Seq[(Generator, Boolean)] = {
```

Comment:
Could you describe the output as a function description, please? For example, the purpose of the second boolean in the result pair?